vm_method.c

```diff
@@ -406,30 +406,21 @@ static void
 rb_method_definition_release(rb_method_definition_t *def, int complemented)
 {
     if (def != NULL) {
-        const int alias_count = def->alias_count;
-        const int complemented_count = def->complemented_count;
-        VM_ASSERT(alias_count >= 0);
-        VM_ASSERT(complemented_count >= 0);
-
-        if (alias_count + complemented_count == 0) {
-            if (METHOD_DEBUG) fprintf(stderr, "-%p-%s:%d,%d (remove)\n", (void *)def,
-                                      rb_id2name(def->original_id), alias_count, complemented_count);
+        const int reference_count = def->reference_count;
+        def->reference_count--;
+
+        VM_ASSERT(reference_count >= 0);
+
+        if (reference_count == 0) {
```

Comment:
Why are we checking the original `reference_count`? Shouldn't we be checking the updated `def->reference_count` here?